### PR TITLE
Added react course of coder dost for beginners in hindi language

### PR DESCRIPTION
**Pull request by pratikkumar399**
## What does this PR do?
Add resource

### Description
React tutorial for beginners in hindi.

### Why is this valuable (or not)?
This is very good react course that is from scratch.

### How do we know it's really free?
Because it is available on youtube for free.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a course on youtube.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-hi.md [hunk 1]
```diff
@@ -432,6 +432,7 @@
 
 #### React
 
+* [Learn React JS in Hindi](https://www.youtube.com/watch?v=6l8RWV8D-Yo&list=PL2PkZdv6p7ZnS2QIz8WAPmqeqUUccPNQY&index=3&t=17419s) - Coder Dost
```
Comment (eshellman) [inline courses/free-courses-hi.md:435]: Don't invent titles! Use the title given by the resource.

Comment (pratikkumar399) [reply courses/free-courses-hi.md]: Applied the changes

File: courses/free-courses-hi.md [hunk 2]
```diff
@@ -432,6 +432,7 @@
 
 #### React
 
+* [10-Hour React Tutorial 2023 - Zero to Advanced | Learn React JS in Hindi](https://www.youtube.com/watch?v=6l8RWV8D-Yo&list=PL2PkZdv6p7ZnS2QIz8WAPmqeqUUccPNQY&index=3&t=17419s) - Coder Dost
```
Comment (eshellman) [inline courses/free-courses-hi.md:435]:
```suggestion
* [10-Hour React Tutorial 2023 - Zero to Advanced \| Learn React JS in Hindi](https://www.youtube.com/watch?v=6l8RWV8D-Yo&list=PL2PkZdv6p7ZnS2QIz8WAPmqeqUUccPNQY) - Coder Dost
```

**Review by Ajay-singh1 — APPROVED** [review]
LGTM!

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by pratikkumar399** [comment]
@eshellman please add hacktoberfest label if possible

**Pull request merged**